src/Core/Addon/Theme/ThemeManager.php

```diff
@@ -597,15 +603,24 @@ private function getDefaultDomains($locale, $themeProvider)
     }
 
     /**
-     * @param TranslationService $translationService
      * @param MessageCatalogue $messageCatalog
      * @param array $allDomains
-     * @param \PrestaShopBundle\Entity\Lang $lang
      * @param string $locale
      * @param string $themeName
      */
-    private function handleImport(TranslationService $translationService, MessageCatalogue $messageCatalog, $allDomains, $lang, $locale, $themeName)
-    {
+    private function handleImport(
+        MessageCatalogue $messageCatalog,
+        array $allDomains,
+        string $locale,
+        string $themeName
+    ): void {
+        // retrieve Lang doctrine entity
+        try {
+            $lang = $this->translationService->findLanguageByLocale($locale);
+        } catch (Exception $exception) {
+            PrestaShopLogger::addLog('ThemeManager->handleImport() - Locale ' . $locale . ' does not exists');
+            throw new \RuntimeException('Lang associated to locale ' . $locale . ' not found.');
```

Comment:
Should be LocaleNotFoundException instead :thinking:

Reply:
LanguageNotFound you meant ?

Reply:
@PierreRambaud ping :)